lib/options.nix

```diff
@@ -403,5 +403,31 @@ rec {
         if cfg ? lazyLoad then lib.literalMD "`false` when lazy-loading is enabled." else true;
       example = false;
     };
+
+  mkMaybeUnpackagedPackage =
```

Comment:
Could we split this into `mkMaybeUnpackagedPackageOption` and `mkUnpackagedPackageOption`?

That way calling code could decide for itself whether to use `mkPackageOption` or `mkUnpackagedPackageOption` if needed.